chainer_chemistry/links/readout/ggnn_readout.py

```diff
@@ -11,43 +11,31 @@ class GGNNReadout(chainer.Chain):
         out_dim (int): dimension of output feature vector
         hidden_dim (int): dimension of feature vector associated to
             each atom
-        n_layers (int): number of layers
-        concat_hidden (bool): If set to True, readout is executed in
-            each layer and the result is concatenated
         nobias (bool): If ``True``, then this function does not use
             the bias
         activation (~chainer.Function or ~chainer.FunctionNode):
             activate function
             It can be replaced with the functions.identity.
     """
 
-    def __init__(self, out_dim, hidden_dim=16, n_layers=4,
-                 concat_hidden=False, nobias=False,
+    def __init__(self, out_dim, hidden_dim=16, nobias=False,
                  activation=functions.tanh):
         super(GGNNReadout, self).__init__()
-        n_layer = n_layers if concat_hidden else 1
         with self.init_scope():
-            self.i_layers = chainer.ChainList(
-                *[GraphLinear(None, out_dim, nobias=nobias)
-                  for _ in range(n_layer)])
-            self.j_layers = chainer.ChainList(
-                *[GraphLinear(None, out_dim, nobias=nobias)
-                  for _ in range(n_layer)])
+            self.i_layer = GraphLinear(None, out_dim, nobias=nobias)
+            self.j_layer = GraphLinear(None, out_dim, nobias=nobias)
         self.out_dim = out_dim
         self.hidden_dim = hidden_dim
-        self.n_layers = n_layers
-        self.concat_hidden = concat_hidden
         self.nobias = nobias
         self.activation = activation
 
-    def __call__(self, h, h0=None, step=0):
+    def __call__(self, h, h0=None):
         # --- Readout part ---
-        index = step if self.concat_hidden else 0
         # h, h0: (minibatch, atom, ch)
         h1 = functions.concat((h, h0), axis=2) if h0 is not None else h
 
-        g1 = functions.sigmoid(self.i_layers[index](h1))
-        g2 = self.activation(self.j_layers[index](h1))
+        g1 = functions.sigmoid(self.i_layer(h1))
+        g2 = self.activation(self.j_layer(h1))
         # sum along atom's axis
         g = self.activation(functions.sum(g1 * g2, axis=1))
```

Comment:
Please remove to apply activation here.

It was not existed.
https://github.com/pfnet-research/chainer-chemistry/blob/v0.4.0/chainer_chemistry/models/ggnn.py#L105

Reply:
According to the paper, https://arxiv.org/abs/1511.05493, tanh functions are applied in eq.7.
It also says that "The tanh functions can also be replaced with the identity".
I think you are able to select which functions to use.

Reply:
I see... in that case I think it is nice to to separate activation function for line 38 and 40, so that one may use tanh in only one place. what do you think?